### ISSUE TITLE
getRegisterable() doesn't always return the registerable domain only

### ISSUE THREAD

**Issue by artyuum** [utopia-php/domains#1]
**Steps to reproduce**
```php
$domain = new Domain('http://facebook.com');
$domain2 = new Domain('http://www.facebook.com');

$domain->getRegisterable()); // 'http://facebook.com'
$domain2->getRegisterable()); // 'facebook.com'
```

**Comment by eldadfux** [comment]
@artyuum please not the library constructor is expecting a domain or hostname as the first argument and not a URL.

**Comment by eldadfux** [comment]
You can try and use it this way:

```php

        $domain = new Domain(parse_url('https://www.facebook.com', PHP_URL_HOST));
       
        $this->assertEquals('www.facebook.com', $domain->get());
        $this->assertEquals('com', $domain->getTLD());
        $this->assertEquals('com', $domain->getSuffix());
        $this->assertEquals('facebook.com', $domain->getRegisterable());
        $this->assertEquals('facebook', $domain->getName());
        $this->assertEquals('www', $domain->getSub());
        $this->assertEquals(true, $domain->isKnown());
        $this->assertEquals(true, $domain->isICANN());
        $this->assertEquals(false, $domain->isPrivate());
        $this->assertEquals(false, $domain->isTest());
        
        $domain = new Domain(parse_url('https://facebook.com', PHP_URL_HOST));
       
        $this->assertEquals('facebook.com', $domain->get());
        $this->assertEquals('com', $domain->getTLD());
        $this->assertEquals('com', $domain->getSuffix());
        $this->assertEquals('facebook.com', $domain->getRegisterable());
        $this->assertEquals('facebook', $domain->getName());
        $this->assertEquals('', $domain->getSub());
        $this->assertEquals(true, $domain->isKnown());
        $this->assertEquals(true, $domain->isICANN());
        $this->assertEquals(false, $domain->isPrivate());
        $this->assertEquals(false, $domain->isTest());
```

**Comment by artyuum** [comment]
Yup, that's the workaround I found in order to strip the "http(s)://" part. The problem is that no error/exception were returned. Why not use `parse_url()` internally? Or maybe this is out of the scope of this library?

**Comment by eldadfux** [comment]
I think detecting if the value is a URL, hostname, or domain might be out of scope, and add unwanted complexity.

I can add a small validation for strings that starts with `https://` or `http://` and throw an exception because this might be a common mistake.

**Comment by artyuum** [comment]
> I can add a small validation for strings that starts with https:// or http:// and throw an exception because this might be a common mistake.

Agreed.

**Comment by eldadfux** [comment]
@artyuum, I just pushed a fix, and released a new tag (0.2.2).

![image](https://user-images.githubusercontent.com/1297371/93639271-66831780-fa01-11ea-94d1-ee9f65bed8f3.png)

**Comment by artyuum** [comment]
Thank you.
